frontend/src/features/admin-form/create/logic/components/LogicContent/EditLogicBlock/EditCondition/ThenShowBlock.tsx

```diff
@@ -70,13 +74,67 @@ export const ThenShowBlock = ({
       : 'show'
   }, [logicTypeValue])
 
+  const [deletedFieldsCount, setDeletedFieldsCount] = useState<number>(0)
+
+  /**
+   * Compute whether any/all fields in the show fields are deleted, then run
+   * effect to delete fields on render and show appropriate error/infobox.
+   */
+  const showValueWatch = useWatchDependency(watch, 'show')
```

Comment:
why do we need to `watch` this? does the component not automatically update when `formFields` updates?

Reply:
`formFields` only contains the fields in the form, not the logics. But we should actually just `watch('show')` here though, will fix.

Reply:
i might be wrong / there might be a better way to achieve this though!